skbio/stats/composition.py

```diff
@@ -1105,6 +1212,34 @@ def _gram_schmidt_basis(n):
     return basis.T
 
 
+def _build_basis(sbp):
+    """
+    Builds an orthogonal basis from a sequential binary partition. 
+       
+    "Code for a SBP. The first row represents the labels of parts of the compositional vector. At each order of
+    partition, +1 means that the part is assigned to the first group, −1 to the second group, and 0 that this part is not
+    in the group which is divided at this order."
+    J.J. Egozcue and V. Pawlowsky-Glahn, http://dugi-doc.udg.edu/bitstream/handle/10256/660/Egozcue-Pawlowsky.pdf
```

Comment:
Nice ascii art! If I understand correctly you already created a way to construct a basis from a SBP with `TreeNode`. Then one can specify this basis in the `ilr` method. If so their is no need for my `_build_basis` method. 

I'm not sure if its the right place to ask the question, but how would you actually transform a SBP to a basis then use it in the `ilr` method?

Thanks